bluesky_queueserver/manager/manager.py

```diff
@@ -1032,12 +1066,114 @@ async def _queue_item_add_handler(self, request):
+
+        success, msg, item_list, qsize = True, "", [], None
+
+        try:
+            # Prepare items
+            if "items" not in request:
+                raise Exception("Invalid request format: the list of items is not found")
+            items = request["items"]
+            items_prepared, items_added, report, success = [], [], [], True
+
+            user, user_group = self._get_user_info_from_request(request=request)
+
+            # First validate all the items
+            for item_info in items:
+                item, item_type = {}, None
+                try:
+                    item, item_type = self._get_item_from_request(request=item_info)
+
+                    # Always generate a new UID for the added plan!!!
+                    item_prepared, _ = self._prepare_item(
+                        item=item, item_type=item_type, user=user, user_group=user_group, generate_new_uid=True
+                    )
+
+                    items_prepared.append(item_prepared)
+                    report.append({"success": True, "msg": ""})
+
+                except Exception as ex:
+                    success = False
+                    # If 'item_info' does not contain valid item, then do not return any item.
+                    if item and item_type:
+                        item["item_type"] = item_type
+                    items_prepared.append(item)  # Always add item even if it is '{}'
+                    report.append({"success": False, "msg": f"Failed to add a plan: {ex}"})
+
+            if len(report) != len(items) != len(items_prepared):
+                # This error should never happen, but it may be useful for debugging.
+                raise Exception("Error in data processing algorithm occurred")
+
+            if success:
+                # Adding plan to queue may still raise an exception
+                for item in items_prepared:
+                    item_added, _ = await self._plan_queue.add_item_to_queue(item)
+                    items_added.append(item_added)
+            else:
+                items_added = items_prepared
+
+            # Prepare the list of added items with error messages
+            for item, rep in zip(items_added, report):
+                d = rep.copy()
+                if "item_type" in item:
+                    d[item["item_type"]] = item
+                item_list.append(d)
+
+            if not success:
+                n_items = len(report)
+                n_failed = sum([not _["success"] for _ in report])
+                msg = f"Failed to add all items: validation of {n_failed} out of {n_items} submitted items failed"
+
+        except Exception as ex:
+            success = False
+            msg = f"Failed to add an item: {str(ex)}"
+
+        try:
+            qsize = await self._plan_queue.get_queue_size()
+        except Exception:
+            qsize = None
+
+        logger.info(self._generate_item_log_msg("Batch of items added", success, None, None, qsize))
```

Comment:
Wouldn't it be better to use `batch` instead of the first `None`?

Reply:
I am not sure. `item_type` should be `plan` or `instruction`. For the batch, `item_type` is set for each item in the list. In the log this statement will be preceded with `"Adding a batch of items to the queue ..."`. The statement itself mentions that the batch is added.